File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int result = nums[0];
        int total = 0;

        for (int i = 0; i < nums.length; i++) {
            if (total < 0)
                total = 0;

            total += nums[i];
            if (total > result)
                result = total;
        }
        return result;
    }
}

File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = 0, n = nums.size();
        int max = nums[0];
        for (int i = 0; i < n; i++) {
            sum += nums[i];
            max = (max < sum) ? sum : max;
            if (sum < 0)
                sum = 0;
        }
        return max;
    }
};